lib/prometheus/business_collector.rb

```diff
@@ -99,17 +105,22 @@ def collect
       @client_systems_by_os.observe(item['count'], version: item['version'])
     end
 
-    hosts_50_plus = PolicyHost.joins(:policy)
-                              .select('policies.account_id')
-                              .group(:policy_id, 'policies.account_id')
-                              .having('COUNT(policy_hosts.host_id) >= 50')
+    policies_50plus_hosts = PolicyHost.joins(:policy)
+                                      .select('policies.account_id')
+                                      .group(:policy_id, 'policies.account_id')
+                                      .having('COUNT(policy_hosts.host_id) >= 50')
+
+    @total_accounts_with_50plus_hosts_per_policy.observe(
+      Account.where(id: policies_50plus_hosts).count(:account_number)
+    )
 
-    @total_accounts_with_huge_policies.observe(
-      Account.where(id: hosts_50_plus).count(:account_number)
+    @client_accounts_with_50plus_hosts_per_policy.observe(
+      client_accounts.where(id: policies_50plus_hosts).count(:account_number)
     )
 
-    @client_accounts_with_huge_policies.observe(
-      client_accounts.where(id: hosts_50_plus).count(:account_number)
+    @total_50plus_policies.observe(policies_50plus_hosts.count.size)
```

Comment:
What's up with `count.size`?

https://ruby-doc.org/core-2.5.0/Integer.html#method-i-size

> Returns the number of bytes in the machine representation of int (machine dependent).

Reply:
@akofink in some cases `count` behaves weird, it returns a hash :confused:

Reply:
count will show the counts of each group when you use `group`. I believe you'll get what you want if you pass the single column you want to count, like `count(:policy_id)`.

Reply:
No, it unfortunately doesn't work like that :confused: whatever I pass to the `count` it will return a hash :confused: because it's not being `SELECT`-ed, but we can't add anything to the `SELECT` above because then the query just won't work :shrug:

Reply:
Indeed, the `group` changes the `count` method's behavior. I guess it is okay to leave the size there if we want to count number of groups.
https://guides.rubyonrails.org/active_record_querying.html#total-of-grouped-items

Reply:
Gotcha! Sorry, I didn't realize we _wanted_ to count the groups :+1: I also could have tested it to find out (sorry about that)